src/config.rs

```diff
@@ -32,34 +33,79 @@ impl FromStr for WorkMode {
 }
 
 #[derive(Debug)]
-pub enum ConfigError {
+pub enum Error {
+    CommandLine(DocoptError),
     Io(IoError, String),
-    TomlError(String, String),
+    Toml(String, String),
+    Options(String, String),
 }
 
-impl Error for ConfigError {
+impl StdError for Error {
     fn description(&self) -> &str {
+        use self::Error::*;
         match *self {
-            ConfigError::Io(ref e, _) => e.description(),
-            ConfigError::TomlError(ref s, _) => s,
+            CommandLine(ref e) => e.description(),
+            Io(ref e, _) => e.description(),
+            Toml(ref s, _) => s,
+            Options(ref s, _) => s,
         }
     }
 }
 
-impl Display for ConfigError {
+impl Display for Error {
     fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        use self::Error::*;
         match *self {
-            ConfigError::Io(ref err, ref filename) => {
-                try!(write!(f, "\"{}\": ", filename));
+            CommandLine(ref err) => err.fmt(f),
+            Io(ref err, ref filename) => {
+                try!(write!(f, "Failed to read config \"{}\": ", filename));
                 err.fmt(f)
             }
-            ConfigError::TomlError(ref err, ref filename) => {
+            Toml(ref err, ref filename) => {
+                write!(f, "\"{}\": {}", filename, err)
+            }
+            Options(ref err, ref filename) => {
                 write!(f, "\"{}\": {}", filename, err)
             }
         }
     }
 }
 
+impl<'a> From<DocoptError> for Error {
+    fn from(e: DocoptError) -> Error {
+        Error::CommandLine(e)
+    }
+}
+
+impl<'a> From<(IoError, &'a str)> for Error {
+    fn from(e: (IoError, &'a str)) -> Error {
+        Error::Io(e.0, e.1.into())
+    }
+}
+
+impl<'a, 'b> From<(&'a str, &'b str)> for Error {
+    fn from(e: (&'a str, &'b str)) -> Error {
+        Error::Options(e.0.into(), e.1.into())
+    }
+}
+
+impl<'a> From<(String, &'a str)> for Error {
+    fn from(e: (String, &'a str)) -> Error {
+        Error::Options(e.0, e.1.into())
+    }
+}
+
+trait LookupStr {
+    fn lookup_str<'a>(&'a self, option: &'a str, err: &str, config_file: &str) -> Result<&'a str, Error>;
+}
+
+impl LookupStr for toml::Value {
+    fn lookup_str<'a>(&'a self, option: &'a str, err: &str, config_file: &str) -> Result<&'a str, Error> {
+        let value = try!(self.lookup(option).ok_or((err, config_file)));
+        Ok(value.as_str().unwrap())
```

Comment:
Perhaps `` "Invalid `{}` value, expected a string, found {}", option, value.type_str() ``
(BTW 'option' is a misleading name for a toml entry, or field, anyway)